treetop.js

```diff
@@ -325,6 +418,28 @@ window.treetop = (function ($, BodyComponent, FormSerializer) {
     PARTIAL_CONTENT_TYPE: "application/x.treetop-html-partial+xml",
     FRAGMENT_CONTENT_TYPE: "application/x.treetop-html-fragment+xml",
 
+    START: "treetopstart",
+    COMPLETE: "treetopcomplete",
+
+    startRequest: function () {
+        this.activeCount++;
+        if (this.activeCount === 1) {
+            var event = document.createEvent("Event");
+            event.initEvent(this.START, false, false);
+            document.dispatchEvent(event);
+        }
+    },
+
+    endRequest: function () {
+        this.activeCount--;
+        if (this.activeCount < 1) {
```

Comment:
Can this just be activeCount == 0? 
When could it be negative

Reply:
It shouldn't ever be. Probably a misguided attempt at being defensive.